packages/core/lib/console.js

```diff
@@ -267,20 +267,31 @@ class Console extends EventEmitter {
       return callback(error);
     }
 
-    // Ensure our script returns a promise whether we're using an
-    // async function or not. If our script is an async function,
-    // this will ensure the console waits until that await is finished.
-    Promise.resolve(runScript(script))
-      .then(value => {
-        // If there's an assignment to run, run that.
-        if (assignment) return runScript(vm.createScript(assignment));
-        return value;
-      })
-      .then(value => {
-        // All good? Return the value (e.g., eval'd script or assignment)
-        callback(null, value);
-      })
-      .catch(callback);
+    const runScriptResult = runScript(script);
+
+    if (runScriptResult instanceof Promise) {
+      // Ensure our script returns a promise whether we're using an
+      // async function or not. If our script is an async function,
+      // this will ensure the console waits until that await is finished.
+      Promise.resolve(runScriptResult)
+        .then(value => {
+          // If there's an assignment to run, run that.
+          if (assignment) return runScript(vm.createScript(assignment));
+          return value;
+        })
+        .then(value => {
+          // All good? Return the value (e.g., eval'd script or assignment)
+          runScript(
+            vm.createScript(
+              'console.log("Entered expression evaluates to promise; auto-resolving:")'
```

Comment:
Can we change the text a bit here? How about "The expression evaluates to a Promise; ..."?